plasmapy/formulary/parameters.py

```diff
@@ -342,6 +344,11 @@ def ion_sound_speed(T_e,
     <Quantity 229585.96150738 m / s>
 
     """
+    
+    if (n_e is None and k is not None) or (k is None and n_e is not None):
+        raise UserWarning("The non-dispersive limit has been assumed for this "
+                          "calculation. To prevent this, values must be "
+                          "specified for both n_e and k.")
```

Comment:
This will interrupt execution (as with an exception), with warnings you need `warn` from the `warnings` module, rather than `raise`!

Also, we could probably throw a `plasmapy.utils.PhysicsWarning` - we've done that for other approximations elsewhere.

Reply:
Ah... ok! I've not used warnings before, so had assumed it would be the same as an exception. Thanks for spotting that! I'll go ahead and change that now.

Reply:
I've fixed the warning (using `warnings.warn`), and changed it from `UserWarning` to `PhysicsWarning`.